starlark/library.go

```diff
@@ -888,6 +889,32 @@ func print(thread *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error
 	return None, nil
 }
 
+// TODO: add link to spec.md
+func fail(thread *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error) {
+	if len(kwargs) > 0 {
+		return nil, fmt.Errorf("fail does not accept keyword arguments")
+	}
+	if len(args) == 0 {
+		return None, fmt.Errorf("none")
+	}
+	attr := ""
+	if err := UnpackPositionalArgs("fail", args, kwargs, 0, &args[0], &attr); err != nil {
```

Comment:
Use a new variable instead of overwriting args[0]. If you make that variable a string, you don't need the check on L909. For example:

var msg, attr string
if err := UnpackPositionalArgs("fail", args, kwargs, 1, &msg, &attr); err != nil {

Reply:
Again, I included args[0] because java's implementation accepted the first arg as any type not just strings.